Doc/library/typing.rst

```diff
@@ -1230,6 +1232,103 @@ These are not used in annotations. They are building blocks for creating generic
+
+    (In older versions of Python, where ``*`` was less flexible, you might see
+    this written using :data:`Unpack <Unpack>` instead, as
+    ``Unpack[Shape]``. This means the same thing as ``*Shape`` - ``Unpack`` and
+    ``*`` can be used interchangeably in the context of types.)
+
+
+    Type variable tuples must *always* be used in combination with the unpacking
+    operator. This helps distinguish type variable types from normal type
+    variables::
+
+        class Array(Generic[Shape]): ...  # Not valid
+
+    In cases where you really do just want a tuple of types, make this explicit
+    by wrapping the type variable tuple in ``tuple[]``::
+
+        shape: Shape          # Not valid
+        shape: tuple[Shape]   # Also not valid
+        shape: tuple[*Shape]  # The correct way to do it
+
+    Type variable tuples can be used in the same contexts as vanilla type
+    variables. For example, in class definitions, arguments, and return types::
+
+        class Array(Generic[*Shape]):
+            def __getitem__(self, key: tuple[*Shape]) -> float: ...
+            def __abs__(self) -> Array[*Shape]: ...
+            def get_shape(self) -> tuple[*Shape]: ...
+
+    Type variable tuples can be happily combined with vanilla type variables::
+
+        DType = TypeVar('DType')
+
+        class Array(Generic[DType, *Shape]):  # This is fine
+            pass
+
+        float_array_1d: Array[float, Height] = Array()     # Totally fine
+        int_array_2d: Array[int, Height, Width] = Array()  # Yup, fine too
+
+    However, note that at most one type variable tuple may appear in a single
+    list of type arguments or type parameters::
+
+        class Array(Generic[*Shape, *Shape]):  # Not valid
+            pass
+
+    Finally, an unpacked type variable tuple can be used as the type annotation
+    of ``*args``::
+
+        def args_to_tuple(*args: *Ts) -> tuple[*Ts]:
+            return tuple(args)
+
+    In contrast to non-unpacked annotations of ``*args`` - e.g. ``*args: int``,
+    which would specify that *all* arguments are ``int`` - ``*args: *Ts``
+    enables reference to the types of the *individual* arguments in ``*args``.
+    In this case, it binds those types to ``Ts``.
+
+    For more details on type variable tuples, see :pep:`646`.
+
+    .. versionadded:: 3.11
+
+.. data:: Unpack
+
+    A typing operator that conceptually marks an object as having been
+    unpacked. For example, using the unpack operator ``*`` on a
+    :class:`type variable tuple <TypeVarTuple>` internally uses ``Unpack``
+    to mark the type variable tuple as having been unpacked::
+
+        Ts = TypeVarTuple('Ts')
+        array: tuple[*Ts]
+        # Effectively does:
+        array: tuple[Unpack[Ts]]
+
+    In fact, ``Unpack`` can be used interchangeably with ``*`` in the context
+    of types. You might see ``Unpack`` being used explicitly in older versions
+    of Python, where ``*`` couldn't be used in certain places.
```

Comment:
It might be good to be a bit more explicit about the advantages of using `Unpack` on older versions of Python

```suggestion
    of types. You might see ``Unpack`` being used explicitly in older versions
    of Python, where ``*`` couldn't be used in certain places::
    
        from typing_extensions import TypeVarTuple, Unpack
        
        Ts = TypeVarTuple('Ts')
        array1: tuple[*Ts]  # Syntax error on Python <= 3.10!
        array2: tuple[Unpack[Ts]]  # Semantically equivalent, and backwards-compatible
```

Reply:
Good idea. Done.